tests/test_market.py

```diff
@@ -424,3 +426,160 @@ def test_apr(self):
+            },
+            # test 3: 5M target_liquidity; 8% APR;
+            #   6mo duration; 22.186877016851916 time_stretch (targets 5% APR);
+            #   1 init share price; 1 share price
+            {
+                "target_liquidity": 5_000_000,  # Targeting 5M liquidity
+                "target_apr": 0.08,  # fixed rate APR you'd get from purchasing bonds; r = 0.08
+                "position_duration": time_utils.StretchedTime(
+                    days=182.5,
+                    time_stretch=13.866798135532449,
+                    normalizing_constant=182.5,
+                ),
+                "init_share_price": 1,  # original share price pool started; u = 1
+                "share_price": 1,  # share price of the LP in the yield source; c = 1
+                "expected_share_reserves": 5_000_000.0,
+                "expected_bond_reserves": 1_806_633.2221533637,
+            },
+            # test 4:  10M target_liquidity; 3% APR
+            #   3mo duration; 36.97812836141986 time_stretch (targets 3% APR);
+            #   1.5 init share price; 2 share price
+            {
+                "target_liquidity": 10_000_000,  # Targeting 10M liquidity
+                "target_apr": 0.03,  # fixed rate APR you'd get from purchasing bonds; r = 0.03
+                "position_duration": time_utils.StretchedTime(
+                    days=91.25,
+                    time_stretch=36.97812836141987,
+                    normalizing_constant=91.25,
+                ),
+                "init_share_price": 2,  # original share price when pool started
+                "share_price": 2,  # share price of the LP in the yield source
+                "expected_share_reserves": 5_000_000.0,
+                "expected_bond_reserves": 1_591_223.795848793,
+            },
+            # test 5:  10M target_liquidity; 5% APR
+            #   9mo duration; 36.97812836141986 time_stretch (targets 3% APR);
+            #   1.5 init share price; 2 share price
+            {
+                "target_liquidity": 10_000_000,  # Targeting 10M liquidity
+                "target_apr": 0.001,  # fixed rate APR you'd get from purchasing bonds; r = 0.03
+                "position_duration": time_utils.StretchedTime(
+                    days=273.75,
+                    time_stretch=1109.3438508425959,
+                    normalizing_constant=273.75,
+                ),
+                "init_share_price": 1.3,  # original share price when pool started
+                "share_price": 1.3,  # share price of the LP in the yield source
+                "expected_share_reserves": 7_692_307.692307692,
+                "expected_bond_reserves": 6_486_058.016848019,
+            },
+        ]
+        # Loop through the test cases & pricing model
+        for test_index, test_case in enumerate(test_cases):
+            # Check if this test case is supposed to fail
+            if "is_error_case" in test_case and test_case["is_error_case"]:
```

Comment:
There are no error cases, I'd say remove this or add some?   `initialize_market` takes `contribution` and `target_apr` so we should make sure that zero liquidity, zero apr error cases are covered.

Reply:
gonna mark this as resolved and do this myself so we can land